pkg/ingester/index/index.go

```diff
@@ -190,12 +239,29 @@ func intersect(a, b []model.Fingerprint) []model.Fingerprint {
 	return result
 }
 
+// hashFP simply moves entropy from the most significant 48 bits of the
+// fingerprint into the least significant 16 bits (by XORing) so that a simple
+// MOD on the result can be used to pick a mutex while still making use of
+// changes in more significant bits of the fingerprint. (The fast fingerprinting
+// function we use is prone to only change a few bits for similar metrics. We
+// really want to make use of every change in the fingerprint to vary mutex
+// selection.)
+func hashFP(fp model.Fingerprint) uint {
```

Comment:
`hashFP()` is now duplicated between `index` and `ingester` packages - could it be moved into a common area?

Reply:
Move to `util`.